File: src/go/sorting/03-insertionSort/InsertionSort.go
package main

import "fmt"

func insertionSort(arr []int) {
    length := len(arr)
    for i := 1; i < length; i++ {
        temp := arr[i]
        j := i - 1
        for j >=0 && arr[j] > temp {
           arr[j+1] = arr[j]
           j = j - 1
        }
        arr[j+1] = temp
    }
}

func main() {
    sortArr := []int{40, 50, 20, 0, -10, 30, 10}
    insertionSort(sortArr)
    fmt.Println(sortArr)
}

File: src/javascript/sorting/insertionSort/insertionSort.js
function insertionSort(array) {
    for(let i = 1; i < array.length; i++) {
        let temp = array[i];
        for(var j = i-1; array[j] > temp && j > -1; j--) {
            array[j+1] = array[j]
        }
        array[j+1] = temp;
    }
    return array;
}

let myArray = [5, 1, 6, 2, 4, 3];
insertionSort(myArray);
console.log(myArray);